### PR TITLE
validate cloud-provider config at startup & make PVProvider optional

### PR DESCRIPTION
**Pull request by skriss**
- validate cloud-provider configurations at server start-up by having the the CP constructors make an API call to confirm the location/AZ is valid

@ncdc PTAL

### CODE REVIEW

File: pkg/cloudprovider/gcp/storage_adapter.go [hunk 1]
```diff
@@ -44,16 +46,24 @@ func NewStorageAdapter(project string, zone string) (cloudprovider.StorageAdapte
...
 	}
 
+	// validate project & zone
+	res, err := gce.Zones.Get(project, zone).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	if res == nil {
+		return nil, fmt.Errorf("zone not found for project %s: %s", project, zone)
```
Comment (ncdc) [inline pkg/cloudprovider/gcp/storage_adapter.go]: How about `zone %q not found for project %q`?

File: pkg/controller/gc_controller.go [hunk 2]
```diff
@@ -115,6 +115,12 @@ func (c *gcController) cleanBackups() {
 			}
 
 			for _, volumeBackup := range backup.Status.VolumeBackups {
+				if c.snapshotService == nil {
+					glog.Warningf("Cannot garbage-collect snapshot %s for %backup s/%s because server is not configured with PersistentVolumeProvider",
```
Comment (ncdc) [inline pkg/controller/gc_controller.go]: typo: `%backup s/%s`

If the snapshot service is nil, the net result is that backups will be GC'd but associated snapshots will be orphaned?

Comment (skriss) [inline pkg/controller/gc_controller.go]: yes. it's a strange situation to be in - you created backups with snapshots, meaning you had a snapshotservice configured, but now you're in a mode where if you restore from them you won't use them. i suppose the alternative is to not GC any backups with snapshots in this scenario - WDYT?

Comment (ncdc) [inline pkg/controller/gc_controller.go]: Maybe let's split out this last commit into a different PR so we can get the config validation in while we continue to discuss the optional PVProvider?

**Comment by skriss** [comment]
Removed commit to make PVProvider optional; will submit in a separate PR.

Changed error messages per comment.

**Comment by ncdc** [comment]
Can you squash the last 2 commits together and then we'll merge?

**Comment by skriss** [comment]
squashed.
